### PR TITLE
The Object class has been renamed to StatementObject

### PR DESCRIPTION
**Pull request by djoudi**

### CODE REVIEW

**Comment by djoudi** [comment]
@xabbuh

**Comment by xabbuh** [comment]
In `composer.json` we need to replace `"php-xapi/model": "^1.0 || ^2.0"` with `"php-xapi/model": "^1.0 || ^2.0 || 3.0"`.

File: src/Normalizer/ObjectNormalizer.php [hunk 1]
```diff
@@ -78,7 +78,7 @@ public function normalize($object, $format = null, array $context = array())
      */
     public function supportsNormalization($data, $format = null)
     {
-        return $data instanceof Object;
+        return $data instanceof StatementObject;
```
Comment (xabbuh) [inline src/Normalizer/ObjectNormalizer.php]: For backwards compatibility, this should be `return $data instanceof Object || $data instanceof StatementObject;`.

File: src/Normalizer/ObjectNormalizer.php [hunk 2]
```diff
@@ -110,7 +110,7 @@ public function denormalize($data, $class, $format = null, array $context = arra
      */
     public function supportsDenormalization($data, $type, $format = null)
     {
-        return 'Xabbuh\XApi\Model\Activity' === $type || 'Xabbuh\XApi\Model\Object' === $type || 'Xabbuh\XApi\Model\StatementReference' === $type || 'Xabbuh\XApi\Model\SubStatement' === $type;
+        return 'Xabbuh\XApi\Model\Activity' === $type || 'Xabbuh\XApi\Model\StatementObject' === $type || 'Xabbuh\XApi\Model\StatementReference' === $type || 'Xabbuh\XApi\Model\SubStatement' === $type;
```
Comment (xabbuh) [inline src/Normalizer/ObjectNormalizer.php]: here we need to check for the old class name too

File: src/Normalizer/ObjectNormalizer.php [hunk 3]
```diff
@@ -128,7 +128,7 @@ private function denormalizeSubStatement(array  $data, $format = null, array $co
     {
         $actor = $this->denormalizeData($data['actor'], 'Xabbuh\XApi\Model\Actor', $format, $context);
         $verb = $this->denormalizeData($data['verb'], 'Xabbuh\XApi\Model\Verb', $format, $context);
-        $object = $this->denormalizeData($data['object'], 'Xabbuh\XApi\Model\Object', $format, $context);
+        $object = $this->denormalizeData($data['object'], 'Xabbuh\XApi\Model\StatementObject', $format, $context);
```
Comment (xabbuh) [inline src/Normalizer/ObjectNormalizer.php]: same here

File: .travis.yml [hunk 4]
```diff
@@ -19,6 +22,10 @@ matrix:
       env: deps="low"
     - php: 7.0
       env: xdebug="yes"
+      php: 7.1
+      env: xdebug="yes"
```
Comment (xabbuh) [inline .travis.yml]: using Xdebug for the code coverage on one job is enough, I would just remove it for the new entries and keep it for the PHP 7.0 job

**Review by xabbuh — CHANGES_REQUESTED** [review]

File: src/Normalizer/ObjectNormalizer.php [hunk 5]
```diff
@@ -128,7 +128,8 @@ private function denormalizeSubStatement(array  $data, $format = null, array $co
     {
         $actor = $this->denormalizeData($data['actor'], 'Xabbuh\XApi\Model\Actor', $format, $context);
         $verb = $this->denormalizeData($data['verb'], 'Xabbuh\XApi\Model\Verb', $format, $context);
-        $object = $this->denormalizeData($data['object'], 'Xabbuh\XApi\Model\Object', $format, $context);
+        if (class_exists('Xabbuh\XApi\Model\Object')) {
+        $object = (class_exists('Xabbuh\XApi\Model\Object'))?$this->denormalizeData($data['object'], 'Xabbuh\XApi\Model\Object', $format, $context):$this->denormalizeData($data['object'], 'Xabbuh\XApi\Model\StatementObject', $format, $context);
```
Comment (xabbuh) [inline src/Normalizer/ObjectNormalizer.php]: I would write it this way:

```php
if (class_exists('Xabbuh\XApi\Model\StatementObject')) {
    $object = $this->denormalizeData($data['object'], 'Xabbuh\XApi\Model\StatementObject', $format, $context);
} else {
    $object = $this->denormalizeData($data['object'], 'Xabbuh\XApi\Model\Object', $format, $context);
}
```

File: src/Normalizer/StatementNormalizer.php [hunk 6]
```diff
@@ -83,7 +83,7 @@ public function denormalize($data, $class, $format = null, array $context = arra
         $id = isset($data['id']) ? StatementId::fromString($data['id']) : null;
         $actor = $this->denormalizeData($data['actor'], 'Xabbuh\XApi\Model\Actor', $format, $context);
         $verb = $this->denormalizeData($data['verb'], 'Xabbuh\XApi\Model\Verb', $format, $context);
-        $object = $this->denormalizeData($data['object'], 'Xabbuh\XApi\Model\Object', $format, $context);
+        $object = $this->denormalizeData($data['object'], 'Xabbuh\XApi\Model\StatementObject', $format, $context);
```
Comment (xabbuh) [inline src/Normalizer/StatementNormalizer.php]: same here

**Review by xabbuh — CHANGES_REQUESTED** [review]

**Pull request closed without merging**